dashboard/api_edx_cache.py

```diff
@@ -251,29 +256,30 @@ def update_cached_certificates(cls, user, edx_client):
                     defaults=updated_values
                 )
             # delete anything is not in the current certificates
-            models.CachedCertificate.delete_all_but(user, all_cert_course_ids)
+            models.CachedCertificate.delete_all_but(user, all_cert_course_ids, provider)
             # update the last refresh timestamp
             cls.update_cache_last_access(user, cls.CERTIFICATE)
         # submit a celery task to reindex the user
         tasks.index_users.delay([user.id], check_if_changed=True)
 
     @classmethod
-    def update_cached_current_grades(cls, user, edx_client):
+    def update_cached_current_grades(cls, user, edx_client, provider):
         """
-        Updates cached current grade data.
+        Updates cached current grade data for given courseware provider
 
         Args:
+            provider (str): name of the courseware backend
             user (django.contrib.auth.models.User): A user
             edx_client (EdxApi): EdX client to retrieve enrollments
         Returns:
             None
         """
 
-        course_ids = models.CachedEnrollment.active_course_ids(user)
+        course_ids = models.CachedEnrollment.active_course_ids(user, provider=provider)
 
         # Current Grades are out of date, so fetch new data from edX.
         current_grades = edx_client.current_grades.get_student_current_grades(
-            get_edxorg_social_username(user), course_ids)
+            get_social_username(user, provider), course_ids)
```

Comment:
`get_social_username` could potentially raise an exception here because it may not exist. I'm thinking we should return from this function if that's the case.

It might also be worth changing `get_social_username` to do a `filter(provider=provider).first()` so it returns `None`, rather than handling it via exception catching, but that's obviously more work so I'll leave that to your judgement if it makes things easier here.

Reply:
it looks like the exception is handled already and it returns None if object not found.